httpbin/helpers.go

```diff
@@ -159,11 +166,29 @@ func parseBody(w http.ResponseWriter, r *http.Request, resp *bodyResponse) error
 		if err != nil && err != io.EOF {
 			return err
 		}
+	// in case of binary data, we encode it as base64 and return it as a data url
+	case strings.HasPrefix(ct, "image/png"):
+		resp.Data = encodeData(body, "image/png")
+	case strings.HasPrefix(ct, "image/jpeg"):
+		resp.Data = encodeData(body, "image/jpeg")
+	case strings.HasPrefix(ct, "image/webp"):
+		resp.Data = encodeData(body, "image/webp")
+	case strings.HasPrefix(ct, "application/octet-stream"):
+		resp.Data = encodeData(body, "application/octet-stream")
+	default:
+		// If we don't know how to handle the content type, we'll just return it encoded as base64 data url
+		resp.Data = encodeData(body, ct)
```

Comment:
Are there any downsides to simplifying all of this down to something like so?

```go
// for any other content-type, encode incoming request body as base64
// and return it as a data url
default:
    resp.Data = encodeData(body, ct)
```

I _think_ we'd need to handle content types like `text/html; charset=utf-8`, where we'd only want to echo back the part before the first `;` (if any).  I suppose that could be done directly in `encodeData()`, or it could be done before this switch statement starts (which would also potentially simplify each of the other cases by eliminating the need for `strings.HasPrefix` for most of them).

Reply:
good point, that makes the code much clearer. I'll push a commit that does just that.